dlp/api/DlpSample/RiskAnalysis.cs

```diff
@@ -0,0 +1,501 @@
+
+    [Verb("lDiversity", HelpText = "Computes the k-anonymity of a column set in a Google BigQuery table.")]
+    abstract class LDiversityOptions : QuasiIdOptions
+    {
+        [Value(7, HelpText = "The column to measure l-diversity relative to. (e.g. 'age')", Required = true)]
+        public string SensitiveAttribute { get; set; }
+    }
+
+    [Verb("kMap", HelpText = "Computes the k-map risk estimation of a column set in a Google BigQuery table.")]
+    abstract class KMapOptions : QuasiIdOptions
+    {
+        [Value(7, HelpText = "A list of the infoTypes for each quasi-id, delimited by commas.", Required = true)]
+        public string InfoTypes { get; set; }
+
+        [Value(8, HelpText = "The ISO 3166-1 region code that the data is representative of.", Default = "")]
+        public string RegionCode { get; set; }
+    }
+
+    public class RiskAnalysis
+    {
+        static object NumericalStats(NumericalStatsOptions opts) {
+            DlpServiceClient dlp = DlpServiceClient.Create();
+
+            // Construct + submit the job
+            RiskAnalysisJobConfig config = new RiskAnalysisJobConfig
+            {
+                PrivacyMetric = new PrivacyMetric {
+                    NumericalStatsConfig = new NumericalStatsConfig {
+                        Field = new FieldId { Name = opts.ColumnName }
+                    }
+                },
+                SourceTable = new BigQueryTable {
+                    ProjectId = opts.TableProjectId,
+                    DatasetId = opts.DatasetId,
+                    TableId = opts.TableId
+                }
+            };
+            config.Actions.Add(new Google.Cloud.Dlp.V2.Action
+            {
+                PubSub = new PublishToPubSub{
+                    Topic = opts.TopicId
+                }
+            });
+
+            var submittedJob = dlp.CreateDlpJob(new CreateDlpJobRequest {
+                Parent = $"projects/{opts.CallingProjectId}",
+                RiskJob = config
+            });
+
+            // Listen to pub/sub for the job
+            SubscriptionName subscriptionName = new SubscriptionName(opts.CallingProjectId,
+    opts.SubscriptionId);
+            SubscriberClient subscriber = SubscriberClient.Create(
+                subscriptionName, new[] { SubscriberServiceApiClient.Create() });
+            // SimpleSubscriber runs your message handle function on multiple
+            // threads to maximize throughput.
+            ManualResetEvent done = new ManualResetEvent(false);
+            subscriber.StartAsync((PubsubMessage message, CancellationToken cancel) =>
+            {
+                if (message.Attributes["DlpJobName"] == submittedJob.Name)
+                {
+                    done.Set();
+                    return Task.FromResult(SubscriberClient.Reply.Ack);
+                }
+                else
+                {
+                    return Task.FromResult(SubscriberClient.Reply.Nack);
```

Comment:
I just read the documentation and used similar code in another sample. It looks like there should be no deadlock because the method implementation returns a task that decided what to do with given message, not with the subscription as I was thinking before.